src/coreclr/vm/exceptionhandling.cpp

```diff
@@ -7559,6 +7563,8 @@ extern "C" void * QCALLTYPE CallCatchFunclet(QCall::ObjectHandleOnStack exceptio
     GCX_COOP_NO_DTOR();
 
     Thread* pThread = GET_THREAD();
+    pThread->DecPreventAbort();
```

Comment:
This handles exception caught by managed code. Do we need to do this when the exception is caught by the unmanaged code?

Reply:
No, we don't
* when an exception crosses managed to native boundary, it still calls the `CallCatchFunclet` with NULL handler. That results in invoking `RealCOMPlusThrow` from the native frame that we've hit. The count is updated properly here. 
* when an exception is thrown by native code and handled by native code without ever passing through managed code, it doesn't need to fiddle with the count, as it never calls the managed EH code